nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/StreamWorkManagerTest.java

```diff
@@ -150,4 +150,21 @@ public void testNoConcurrentJobsWithSameId() throws Exception {
         super.testNoConcurrentJobsWithSameId();
     }
 
+    @Test
+    public void testCoalescingWorks() throws InterruptedException {
+        SleepWork work1 = new SleepWork(getDurationMillis());
+        work1.setIdempotent(false);
+        assertFalse(work1.isIdempotent());
+        work1.setCoalescing(true);
+        assertTrue(work1.isCoalescing());
+        service.schedule(work1);
+        service.schedule(work1);
+        service.schedule(work1);
+        service.schedule(work1);
+        service.schedule(work1);
+        // coalescing works are not executed only
+        assertTrue(service.awaitCompletion(getDurationMillis(), TimeUnit.MILLISECONDS));
+        tracker.assertDiff(0, 0, 5, 0);
+    }
```

Comment:
check your comments the proof is not obvious, might be better if you add a workDuration variable, also how do you know that the first execution is not running before you schedule the others, in this case, it will take at least 2xworkDuration, so you better to assert < 3xworkDuration

Reply:
Actually, even the first one is skipped in this test, the second one is scheduled before the first one is computed.
I will improve the with different durations and some sleeps to prove eveything's fine in both cases (first is computed or not)